datafusion/expr-common/src/columnar_value.rs

```diff
@@ -195,9 +197,12 @@ impl ColumnarValue {
                 kernels::cast::cast_with_options(array, cast_type, &cast_options)?,
             )),
             ColumnarValue::Scalar(scalar) => {
+                // TODO(@notfilippo, logical/physical): this cast in the future should reduce to updating
+                // Scalar's `data_type` to the target `cast_type`.
```

Comment:
Why? Don't we need to transform the actual value, eg from int to string or vice versa?

Reply:
Leftover from wrong notes that I was taking while editing the file. Removed.